File: 2365-percentage-of-letter-in-string/percentage-of-letter-in-string.java
class Solution {
    public int percentageLetter(String s, char letter) {
        HashMap<Character,Integer> mp=new HashMap<>();
        for(char ch:s.toCharArray()){
            mp.put(ch,mp.getOrDefault(ch,0)+1);
        }
        int pl=0;
        for(Map.Entry<Character,Integer> entry:mp.entrySet()){
            if(entry.getKey()==letter){
                pl=entry.getValue();
            }
        }
        int ans=(pl*100)/s.length();
        return ans;
    }
}

File: 2365-percentage-of-letter-in-string/percentage-of-letter-in-string.cpp
class Solution {
public:
    int percentageLetter(string s, char letter) {
        map<char,int> mp;
        for(char ch:s){
            mp[ch]++;
        }
        int pl=0;
        for(auto entry:mp){
            if(entry.first==letter){
                pl=entry.second;
            }
        }
        int ans=(pl*100)/s.length();
        return ans;
    }
};